test/ngAria/ariaSpec.js

```diff
@@ -127,7 +157,7 @@ describe('$aria', function() {
     });
 
     it('should handle checkbox with integer model values using ngTrueValue', function() {
-      var element = $compile('<input type="checkbox" ng-model="val" ng-true-value="0">')(scope);
+      var element = $compile('<div role="checkbox" ng-model="val" ng-true-value="0">')(scope);
```

Comment:
So, after some good time wondering how can this test be passing, I realized that `ng-true-value`/`ng-false-value` is only applied to `input` elements (so it has no effect here). This test only passed by chance.

It would fail for example, by changing `ng-true-value="0"` to `ng-true-value="2"` (and `val=0` to `val=2` below).
I am afraid there is no reliable way of determining when a custom checkbox is selected. I would stick to considering it not selected when `ngModel.$viewValue === false`. That is also consistent with `ngAria`'s `ngModelController.$isEmpty()` function. (See [here][1].)

[1]: https://github.com/angular/angular.js/pull/13483/files#diff-91e9129201746ba107348c9a0a7735edR248